Admin access

hackmastera opened the pull request:
refactor ability spec and demonstrate that admins already have update access to all files. closes #478

jrochkind commented:
This is actually only changes to the _spec_, all that stuff was already working, we just had to verify it in a spec? Great!

jrochkind commented on spec/models/ability_spec.rb:
```diff
@@ -2,42 +2,48 @@
+  before do
+    admin_role.users << admin
+    admin_role.save
+  end
```
If FactoryGirl has a factory defined `:admin`, shouldn't it already be doing this? I see in fact the only difference in the admin factory is the email! 

I think the admin factory should probably be fixed to just do this in the factory. Probably with a find_or_create so it's nice and lazy. 

Otherwise, I'd get rid of the admin factory altogether. And make the user factory use FG's sequence feature, so every one doesn't have the same email, and you can create multiple. 

What do you think?  Just suggestions, we can merge as is depending on your thoughts.

hackmastera commented:
@jrochkind better? Or were you thinking a factory for the admin role, also?

jrochkind commented:
@hackmastera yep, that's exactly what I was thinking! 

Except we probably want all the user factories to use factorygirl sequence feature for email addresses, so it will be possible to create more than one :admin or :user without their emails conflicting and raising a unique violation error (assuming we ensure emails are unique, which hopefully the code does!)

That doesn't seem to be a problem in this test or any existing tests, so up to you whether you want to fix it now (or ask me to :) ), but it's a good practice that will avoid an annoying problem the first time we create a test that does `create(:user)` more than once or whatever.

hackmastera commented:
@jrochkind How about this? A couple looked unused so I just removed them. I changed admin to a trait and got rid of its custom email address. `depositor` is in use and I don't know that I want to spend time messing with that one now; maybe we should just leave it. If this seems good to you and the tests pass I'm inclined to be done for now. Thanks for these suggestions to improve the specs!

hackmastera commented:
(the regular `user` factory is already using sequences)
